homeassistant/components/xiaomi_miio/gateway.py

```diff
@@ -0,0 +1,47 @@
+"""Code to handle a Xiaomi Gateway."""
+import logging
+
+from miio import DeviceException, gateway
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConnectXiaomiGateway:
```

Comment:
This file and could be generalized to allow any sort of xiaomi_miio devices, which will allow us to greatly remove code duplication from all platforms (making other platforms to use this class should go into a separate PR later on).

I'm not sure what would be a good name for the class, but the properties could be renamed:
* gateway_device => device
* info => info